scripts/assert-changed-files.sh

```diff
@@ -1,11 +1,40 @@
 #!/bin/bash
+
+IS_CI="${CI:-false}"
 GREP_PATTERN=$1
+MERGE_SUCCESS=1
+
+
+if [ "$IS_CI" = true ]; then
+  git config --local url."https://github.com/".insteadOf git@github.com:
+  git config --local user.name "GatsbyJS Bot"
+  git config --local user.email "core-team@gatsbyjs.com"
+
+  git fetch origin
+  git merge --no-edit origin/master
+
+  if [ $? -ne 0 ]; then
+    echo "Branch has conflicts with master, rolling back test."
+    MERGE_SUCCESS=0
+    git merge --abort
+  fi
+
+  git config --local --unset user.name
+  git config --local --unset user.email
+  git config --local --unset url."https://github.com/".insteadOf
+fi
 
 FILES_COUNT="$(git diff-tree --no-commit-id --name-only -r "$CIRCLE_BRANCH" origin/master | grep -E "$GREP_PATTERN" -c)"
 
+if [ $MERGE_SUCCESS = 1 ]; then
+  # reset to previous state
+  git reset --hard HEAD@{1}
```

Comment:
What is this `@{1}` doing? just running `git reset --hard HEAD` should be enough

Reply:
I also don't think you need to wrap this in a merge success. It should be a safe no-op to run anytime

Reply:
This is what's happening.

1. git merge from remote/master and commit the changes
2. diff the tree and see what actually changes
3. revert the previous merge commit so were back in PR state.

if you use `git reflog`, you'll see a list of git operations instead of git commits. We can revert the git merge more easily by using `git reset --hard HEAD@{1}`.

You're right that we can always reset it with `HEAD@{1}`